internal/initializer/crds.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// NewCoreCRDs returns a new *CoreCRDs.
+func NewCoreCRDs(path string) *CoreCRDs {
+	return &CoreCRDs{Path: path}
+}
+
+// CoreCRDs makes sure the CRDs are installed.
+type CoreCRDs struct {
+	Path string
+}
+
+// Run applies all CRDs in the given directory.
+func (c *CoreCRDs) Run(ctx context.Context, kube resource.ClientApplicator) error { // nolint:gocyclo
+	var crds []*v1.CustomResourceDefinition
+	err := filepath.Walk(c.Path, func(path string, info os.FileInfo, err error) error {
```

Comment:
@muvaf can we reuse the parser implementation [here](https://github.com/crossplane/crossplane-runtime/blob/30a941c3c3c67e4dc064a2a170cb725de40b0271/pkg/parser/parser.go#L93)? It has support for a filesystem backend that would accommodate this use-case. We use it currently to build the packages in the CLI: https://github.com/crossplane/crossplane/blob/d9afc165cd5335703483b4671b0ca0c751038d63/cmd/crank/build.go#L64

Reply:
I think we can. Let me see.

Reply:
We'd be able to use the backend which reads the files but not the actual parser right? We'd still do the YAML decoding ourselves here?

Reply:
You can use the parser also, you just need to add the proper versions to the `Scheme`. ref: https://github.com/crossplane/crossplane/blob/f6ebabeda19dd31d39adfc71c0c2110ee74032ab/internal/xpkg/scheme.go#L46

Reply:
Oh OK, when I looked at this I saw parser returns a `*Package` struct and thought it's package-specific but it seems it's just two lists of objects. I'll use this implementation.

Reply:
Just pushed a new commit that uses the parser.